File: TinyClaw/Views/Chat/ConfirmationCard.swift
import SwiftUI

/// Inline confirmation card shown when a task or event is created via chat.
struct ConfirmationCard: View {
    let data: Data

    struct Data {
        let title: String
        let subtitle: String  // e.g., "Tomorrow, 5:00 PM" or "March 14, 9:00 AM - 9:30 AM"
        let intent: Intent
    }

    private var accentColor: Color {
        data.intent == .task ? .blue : .purple
    }

    var body: some View {
        HStack(spacing: 8) {
            RoundedRectangle(cornerRadius: 2)
                .fill(accentColor)
                .frame(width: 3)

            VStack(alignment: .leading, spacing: 2) {
                Text(data.title)
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundStyle(accentColor)
                Text(data.subtitle)
                    .font(.system(size: 11))
                    .foregroundStyle(.secondary)
            }

            Spacer()
        }
        .padding(8)
        .background(Color(.systemGray6))
        .clipShape(RoundedRectangle(cornerRadius: 8))
    }
}

File: lib/views/chat/confirmation_card.dart
import 'package:flutter/material.dart' hide Intent;
import '../../core/intent.dart';

/// Inline confirmation card shown when a task or event is created via chat.
class ConfirmationCard extends StatelessWidget {
  final String title;
  final String subtitle;
  final Intent intent;

  const ConfirmationCard({
    super.key,
    required this.title,
    required this.subtitle,
    required this.intent,
  });

  Color get _accentColor => intent == Intent.task ? Colors.blue : Colors.purple;

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.all(8),
      decoration: BoxDecoration(
        color: Theme.of(context).colorScheme.surfaceContainerHighest,
        borderRadius: BorderRadius.circular(8),
      ),
      child: Row(
        children: [
          Container(
            width: 3,
            height: 36,
            decoration: BoxDecoration(
              color: _accentColor,
              borderRadius: BorderRadius.circular(2),
            ),
          ),
          const SizedBox(width: 8),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  title,
                  style: TextStyle(
                    fontSize: 12,
                    fontWeight: FontWeight.w600,
                    color: _accentColor,
                  ),
                ),
                const SizedBox(height: 2),
                Text(
                  subtitle,
                  style: TextStyle(
                    fontSize: 11,
                    color: Theme.of(context).colorScheme.onSurfaceVariant,
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
